### PR TITLE
Fix issue with d3 js tree for tool recommendations

### PR DESCRIPTION
**Pull request by anuprulez**
Fixes d3 JS tree for tool recommendation mentioned in https://github.com/galaxyproject/galaxy/issues/15723

PR against 23.0 on Galaxy main is open (https://github.com/galaxyproject/galaxy/pull/15852), but it may take some time to be merged. Can this PR be merged so that we can test it on the EU? All the lining tests are passing. Only one test fails but that's not related to this PR I think.

ping @bgruening @ElectronicBlueberry  thanks a lot!

### CODE REVIEW

File: client/src/components/ToolRecommendation.vue [hunk 1]
```diff
@@ -86,32 +86,60 @@ export default {
             const duration = 750;
             const maxTextLength = 20;
             const svg = d3.select("#tool-recommendation").append("svg").attr("class", "tree-size").append("g");
-            const gElem = svg[0][0];
+            const gElem = svg._groups[0][0];
```
Comment (ElectronicBlueberry) [inline client/src/components/ToolRecommendation.vue]: Properties prefixed with `_` usually indicate internal properties, which are undocumented, and may change even on minor releases. A quick search through the d3 docs seems to confirm this is also the case here.

There should be an official way to do this, which would be preferable. Could you link to where you found this property?

File: client/src/components/ToolRecommendation.vue [hunk 2]
```diff
@@ -86,32 +86,60 @@ export default {
...
             const clientW = svgElem.clientWidth;
             const translateX = parseInt(clientW * 0.15);
-
             svgElem.setAttribute("viewBox", -translateX + " 0 " + 0.5 * clientW + " " + clientH);
             svgElem.setAttribute("preserveAspectRatio", "xMidYMid meet");
-
-            const tree = d3.tree().size([clientH, clientW]);
-            const diagonal = d3.svg.diagonal().projection((d) => {
-                return [d.y, d.x];
+            var d3Tree = d3.tree().size([clientH, clientW]);
```
Comment (ElectronicBlueberry) [inline client/src/components/ToolRecommendation.vue]:
```suggestion
            const d3Tree = d3.tree().size([clientH, clientW]);
```

File: client/src/components/ToolRecommendation.vue (hunk 1 above)
Comment (anuprulez) [reply client/src/components/ToolRecommendation.vue]: Yes, it also looks weird to me. Currently trying to find its alternative. I found it while looking inside the `svg` object.

File: client/src/components/ToolRecommendation.vue (hunk 2 above)
Comment (anuprulez) [reply client/src/components/ToolRecommendation.vue]: thanks!

File: client/src/components/ToolRecommendation.vue (hunk 1 above)
Comment (mvdbeek) [reply client/src/components/ToolRecommendation.vue]: You could let vue draw the SVG in the template section

Comment (anuprulez) [reply client/src/components/ToolRecommendation.vue]: Using `d3.select(...).node()` offers an easier but effective solution.

**Comment by anuprulez** [comment]
Thanks :)

**Pull request merged**